test-cache/models/Cache-Address.spec.ts

```diff
@@ -0,0 +1,138 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import {expect} from "chai";
+import { CacheAccount } from '../../src/cache-model/cache-account/Cache-Account';
+import { CacheAddress } from '../../src/cache-model/cache-account/Cache-Address';
+import {XEM} from "../../src/models/mosaic/XEM";
+import {NetworkTypes} from "../../src/models/node/NetworkTypes";
+import {MultisigSignatureTransaction} from "../../src/models/transaction/MultisigSignatureTransaction";
+import {EmptyMessage, PlainMessage} from "../../src/models/transaction/PlainMessage";
+import {TimeWindow} from "../../src/models/transaction/TimeWindow";
+import {TransferTransaction} from "../../src/models/transaction/TransferTransaction";
+import {NEMLibrary} from "../../src/NEMLibrary";
+import { TestVariables } from '../../test/config/TestVariables.spec';
+
+declare let process: any;
+
+describe("CacheAccount", () => {
+  const recipientAccount = new CacheAddress(TestVariables.TEST_ADDRESS);
+  const publicKey: string = TestVariables.TEST_PUBLIC_KEY;
+  const privateKey: string = TestVariables.TEST_PRIVATE_KEY;
+  const undefinedPrivateKey: string = process.env.NOT_DEFINED_PRIVATE_KEY;
+
+  before(() => {
+    NEMLibrary.bootstrap(NetworkTypes.TEST_NET);
+  });
+
+  after(() => {
+    NEMLibrary.reset();
+  });
+
+  it("should sign a multisig transaction", () => {
+    const account = new CacheAccount(recipientAccount, publicKey, privateKey);
+    const transaction = new MultisigSignatureTransaction(
+      TimeWindow.createFromDTOInfo(68502384, 68505984),
+      -1744830462,
+      new CacheAddress("TBUAUC3VYKPP3PJPOH7A7BCB2C4I64XZAAOZBO6N"),
+      {data: "d6103d88aad57055e31e8f7a90e08c8a40883b5a427e63e6acc69ccf22ebc94e"},
+      6000000,
+      "585e68a4a4677a2fef3883353e0b1e57d7c9d9a9b0ee84baff58462a044aabee38353c29d81d5ccee23ae1ea6bb34f6bcd54500d4c4d1dd68ce8d0eb6eac730a");
+    const signedTransaction = account.signTransaction(transaction);
+    expect(signedTransaction.data).to.be.equal("021000000200009870431504200000002a9b9786d94b1ce89723c0b4ce1673aaed2185a2d8cee367499d80ac0259f0c6808d5b0000000000805115042400000020000000d6103d88aad57055e31e8f7a90e08c8a40883b5a427e63e6acc69ccf22ebc94e280000005442554155433356594b505033504a504f48374137424342324334493634585a41414f5a424f364e");
+    expect(signedTransaction.signature).to.be.equal("474c081045e38f4b77a9ace17c38cdd1c1f5693a2553e99e0b8fc91168c2fe0837214ccaba0031ca54aa96cd7dfb8d789fcd95d3fa2692429633a3bec8502c00");
+  });
+
+  it("should sign a transaction with message", () => {
```

Comment:
Similar here. I think we aren't adding new value by re-doing these tests perhaps. Maybe a better test is just making sure our subclass properly inherits from the parent class, then we can leave the old tests to test the parent class